src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -411,6 +411,27 @@ private BoundExpression BindSimpleBinaryOperator(BinaryExpressionSyntax node, Di
             int compoundStringLength = 0;
 
             BoundExpression result = BindExpression(current, diagnostics);
+
+            if (node.IsKind(SyntaxKind.SubtractExpression)
+                && current.IsKind(SyntaxKind.ParenthesizedExpression))
+            {
+                if (result.Kind == BoundKind.TypeExpression
+                    && !((ParenthesizedExpressionSyntax)current).Expression.IsKind(SyntaxKind.ParenthesizedExpression))
+                {
+                    Error(diagnostics, ErrorCode.ERR_PossibleBadNegCast, node);
+                }
+                else if (result.Kind == BoundKind.BadExpression)
+                {
+                    var parenthesizedExpression = (ParenthesizedExpressionSyntax) current;
+
+                    if (parenthesizedExpression.Expression.IsKind(SyntaxKind.IdentifierName)
+                        && ((IdentifierNameSyntax) parenthesizedExpression.Expression).Identifier.ValueText == "dynamic")
```

Comment:
@AlekseyTs dropped by to explain to me that @dynamic is another name for the dynamic type (if no type of that name is present). I therefore withdraw my objection.